grate/Migration/SqlServerDatabase.cs

```diff
@@ -14,5 +17,42 @@ public SqlServerDatabase(ILogger<SqlServerDatabase> logger)
         public override bool SupportsDdlTransactions => true;
         public override bool SupportsSchemas => true;
         protected override DbConnection GetSqlConnection(string? connectionString) => new SqlConnection(connectionString);
+
+        public override async Task RestoreDatabase(string restoreFromPath)
+        {
+            try
+            {
+                await OpenAdminConnection();
+                Logger.LogInformation("Restoring {dbName} database on {server} server from path {path}.", DatabaseName, ServerName, restoreFromPath);
+                using var s = new TransactionScope(TransactionScopeOption.Suppress, TransactionScopeAsyncFlowOption.Enabled);
+                var cmd = AdminConnection.CreateCommand();
+                cmd.CommandText =
+                        $@"USE master
+                        ALTER DATABASE [{DatabaseName}] SET SINGLE_USER WITH ROLLBACK IMMEDIATE;
+                        RESTORE DATABASE [{DatabaseName}]
+                        FROM DISK = N'{restoreFromPath}'
+                        WITH NOUNLOAD
+                        , STATS = 10
```

Comment:
I'm not experienced in restoring databases with T-SQL, are these options for STATS, RECOVERY and REPLACE the "standard way" of doing it?

Reply:
I believe these options came up when I took the script that SSMS generated when I restored a db, but I will double check and remove any that are not.

Reply:
RECOVERY did not come up when I just did a restore in SSMS but I believe that is due to how I had things set by default. Based on what I have read [here](https://docs.microsoft.com/en-us/sql/relational-databases/backup-restore/restore-a-database-backup-using-ssms?view=sql-server-ver15#examples), adding with recovery is to indicate there will be no other backups to be restored, instead of leaving it in a state where additional transaction logs can be restored
STATS came up and docs mention it is for reporting back progress of the restore in that percentage increment specified.
NOUNLOAD has to do with restoring from a tape drive, so it is not needed here and will be removed.
REPLACE would be needed here for replacing the existing DB with this backup.

Also, I took a peek at [Roundhouse](https://github.com/chucknorris/roundhouse/blob/master/product/roundhouse.databases.sqlserver/SqlServerDatabase.cs#L204) and could see they restored in SQL Server with NOUNLOAD, STATS, RECOVERY, and REPLACE.